Makefile

```diff
@@ -1,19 +1,23 @@
 TOPDIR := $(PWD)
 PLATFORMS := $(foreach platform,$(wildcard platforms/*),$(platform)/gpt)
 PARTITIONS_XML := $(foreach platform,$(wildcard platforms/*),$(platform)/partitions.xml)
-BINS := gen_partition.py msp.py ptool.py
+CONTENTS_XML := $(patsubst %.xml.in,%.xml,$(wildcard platforms/*/contents.xml.in))
+BINS := gen_contents.py gen_partition.py msp.py ptool.py
 PREFIX ?= /usr/local
 
 .PHONY: all check clean lint integration
 
-all: $(PLATFORMS) $(PARTITIONS_XML)
+all: $(PLATFORMS) $(PARTITIONS_XML) $(CONTENTS_XML)
 
 %/gpt: %/partitions.xml
 	cd $(shell dirname $^) && $(TOPDIR)/ptool.py -x partitions.xml
 
 %/partitions.xml: %/partitions.conf
 	$(TOPDIR)/gen_partition.py -i $^ -o $@
 
+%/contents.xml: %/partitions.xml
+	$(TOPDIR)/gen_contents.py -p $^ -t $@.in -o $@
+
 lint:
 	# W605: invalid escape sequence
 	pycodestyle --select=W605 *.py
```

Comment:
This is only testing for one warning (W605) as to avoid a regression, but since your new script is near perfect, perhaps you should break this in two calls, one for the current scripts and one for yours that would enable almost all checks (except "E501 line too long")?